File: study/lua/src/app/scene/view/settlement/SummarySingleRaceEnd.lua
-- 跨服个人竞技战斗结束

local SummaryBase = require("app.scene.view.settlement.SummaryBase")
local SummarySingleRaceEnd = class("SummarySingleRaceEnd", SummaryBase)

function SummarySingleRaceEnd:ctor(battleData, callback, attackHurt)
    self._battleData = battleData
    local list = {}
    local size = G_ResolutionManager:getDesignCCSize()
    local width = size.width
    local height = size.height

    local midXPos = 0

    local componentSingleRacePlayers = require("app.scene.view.settlement.ComponentSingleRacePlayers").new(battleData, cc.p(midXPos, 265 - height*0.5))
    table.insert(list, componentSingleRacePlayers)

    SummarySingleRaceEnd.super.ctor(self, battleData, callback, list)
end

function SummarySingleRaceEnd:onEnter()
    SummarySingleRaceEnd.super.onEnter(self)
    self:_createAnimation()
end

function SummarySingleRaceEnd:onExit()
    SummarySingleRaceEnd.super.onExit(self)
end


function SummarySingleRaceEnd:_createAnimation()
    local EffectGfxNode = require("app.effect.EffectGfxNode")
    local function effectFunction(effect)
        local subEffect = EffectGfxNode.new(effect)
        subEffect:play()
        return subEffect
    end
    G_EffectGfxMgr:createPlayMovingGfx( self, "moving_battleend", effectFunction, handler(self, self.checkStart) , false )
end

return SummarySingleRaceEnd

File: study/ts/script/scene/view/settlement/SummarySingleRaceEnd.ts
import SummaryBase from "./SummaryBase";
import { G_EffectGfxMgr, G_ResolutionManager } from "../../../init";
import { handler } from "../../../utils/handler";
import ComponentBase from "./ComponentBase";
import ComponentSingleRacePlayers from "./ComponentSingleRacePlayers";

const { ccclass, property } = cc._decorator;
@ccclass
export default class SummarySingleRaceEnd extends SummaryBase {

    private _battleData;
    public init(battleData, callback, attackHurt?) {
        this._battleData = battleData;
        var list: ComponentBase[] = [];

        var height = G_ResolutionManager.getDesignHeight();

        let midXPos = 0;
        var componentSingleRacePlayers = new cc.Node("componentSingleRacePlayers").addComponent(ComponentSingleRacePlayers);
        componentSingleRacePlayers.init(battleData, cc.v2(midXPos, 265 - height*0.5))
        list.push(componentSingleRacePlayers);

        super.init(this._battleData, callback, list);
    }
    public onEnter() {
        super.onEnter();
        this._createAnimation();
    }
    public onExit() {
        super.onExit();
    }
    private _createAnimation() {
        G_EffectGfxMgr.createPlayMovingGfx(this.node, 'moving_battleend', null, handler(this, this.checkStart), false);
    }
}